event/handler.go

```diff
@@ -2,34 +2,42 @@ package event
 
 import (
 	"context"
-	"fmt"
-	"io/ioutil"
 
 	"github.com/ONSdigital/dp-search-reindex-tracker/config"
-	"github.com/ONSdigital/log.go/v2/log"
 )
 
-// TODO: remove hello called example handler
-// HelloCalledHandler ...
-type HelloCalledHandler struct {
+//go:generate mockgen -destination mock/handler.go -package mock github.com/ONSdigital/dp-search-reindex-tracker/event Handler
+
+// Handler represents a handler for processing a single event.
+type Handler[M KafkaAvroModel] interface {
+	Handle(ctx context.Context, cfg *config.Config, topicModel *M) error
+}
+
+// ReindexRequestedHandler is the handler for reindex requested messages.
+type ReindexRequestedHandler struct {
+}
+
+// Handle takes a single event
+func (h *ReindexRequestedHandler) Handle(ctx context.Context, cfg *config.Config, event *ReindexRequestedModel) error {
+	// TO-DO: add job id to in-memory variable for global lookup e.g. map[<job_id>] bool
+	// TO-DO: make a request to the Search Reindex API (via API router) to update state to in-progress for the job
```

Comment:
TODO all one word 
```suggestion
	// TODO: add job id to in-memory variable for global lookup e.g. map[<job_id>] bool
	// TODO: make a request to the Search Reindex API (via API router) to update state to in-progress for the job
```

Reply:
Additionally, if this function has a TODO in it, why don't the other blank handle functions for the other events?

Reply:
I currently don't know how the other events are to be handled. On the other hand, I know of the first one which I will be implementing as my next task